deployment/docker-swarm/README.md

```diff
@@ -0,0 +1,513 @@
+
+```docker-compose.local-overrides.yml``` 
+
+or 
+
+```docker-compose.local.overrides.yml``` and ```blackduck-alert.env```
+
+### Edit Environment File
+Environment variables for Alert have already been created in this file but they are commented out.  
+Uncomment the variable you wish to set by deleting the '#' character at the beginning of each line and set its value.
+
+Example: 
+```
+ALERT_HOSTNAME=localhost
+ALERT_LOGGING_LEVEL=INFO
+```
+
+### Edit the Overrides File
+Uncomment environment from the alert service section of docker-compose.local-overrides.yml.  
+Add environment variables as ```- <VARIABLE_NAME>=<VARIABLE_VALUE>``` into the ```environment: ``` section of the alert service.
+
+Example: 
+```
+    environment:
+        - ALERT_HOSTNAME=localhost
+        - ALERT_LOGGING_LEVEL=INFO
+```
+
+### Environment Variable Overrides
+The environment variables will always take precedence and overwrite the values stored in the database if the following variable value is set to 'true'.
+
+```ALERT_COMPONENT_SETTINGS_SETTINGS_STARTUP_ENVIRONMENT_VARIABLE_OVERRIDE=true```
+
+### Alert Hostname Variable
+You must specify the ALERT_HOSTNAME environment variable in order for Alert to generate and use certificates correctly.
+- Add the ALERT_HOSTNAME environment variable the value must be the hostname only 
+    - Editing environment file:
+        ```
+        ALERT_HOSTNAME=<NEW_HOST_NAME>
+        ```
+    - Editing overrides file:
+        ```
+            environment:
+                - ALERT_HOSTNAME=<NEW_HOST_NAME>
+        ```
+- Do not add the protocol a.k.a scheme to the value of the variable.
+    - Good: ```ALERT_HOSTNAME=myhost.example.com```
+    - Bad: ```ALERT_HOSTNAME=https://myhost.example.com```
+
+### Alert Logging Level Variable
+To change the logging level of alert add the following environment variable to your deployment. 
+
+- Editing environment file: 
+    ```ALERT_LOGGING_LEVEL=DEBUG```
+- Editing overrides file: 
+    ```
+        environment: 
+           - ALERT_LOGGING_LEVEL=DEBUG
+    ```
+
+- Set the value to one of the following: 
+    - DEBUG
+    - ERROR
+    - INFO
+    - TRACE
+    - WARN
+
+### Email Channel Environment Variables
+A majority of the Email Channel environment variables that can be set are related to JavaMail configuration properties.  You can find the JavaMail properties here: 
+
+[JavaMail Properties](https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html)
+
+- The Email Channel environment variables have a prefix of ```ALERT_CHANNEL_EMAIL_```
+- The remaining portion of the variable, after the prefix, map to the JavaMail properties if you replace '_' with '.'
+
+Examples:
+- ALERT_CHANNEL_EMAIL_MAIL_SMTP_HOST maps to 'mail.smtp.host'
+- ALERT_CHANNEL_EMAIL_MAIL_SMTP_PORT maps to 'mail.smtp.port'
+
+### Environment Variable Classifications
+There are certain classifications with the environment variables. The variables have a specific naming convention:
```

Comment:
Does this seem redundant to you? I feel this can be expressed in a single sentence.